doc/tutorials/distributed.rst

```diff
@@ -4,6 +4,105 @@ Distributed Secure XGBoost
 
 Secure XGBoost is horizontally scalable -- if you want to train on more data, you can simply add more machines to the enclave cluster. Of course, the presence of an enclave cluster means that training will happen in a distributed manner. XGBoost itself already supports distributed training, and this project has adopted the native distributed training/inference logic and made it more secure.
 
-In the distributed setting, nodes logically communicate with each other in either a tree or ring structure (depending on the data size), with one node designated as the master. For security purposes in the :doc: `outsourced computation model </outsourced>`, the client should authenticate each enclave and verify that it's running the proper code by attesting each enclave in the cluster. However, to simplify the process for the client, Secure XGBoost leverages inter-enclave attestation -- when the cluster is initialized, each enclave attests its neighbors. Therefore, instead of having to attest each enclave separately, the client only needs to attest the master. 
-
 Additionally, all communication within the cluster is encrypted and happens over TLS. Each communication channel begins and ends inside an enclave, so data communicated between enclaves are never exposed to an untrusted host. 
+
+
+**Contents**
+
+* `Architecture`_
+
+* `Distributed Training`_
+
+* `Tutorial`_
+
+* `Troubleshooting`_
+
+************
+Architecture
+************
+Distributed Secure XGBoost leverages a tracker, or an orchestrator, to configure the cluster. This involves setting up the cluster topology, i.e. the communication patterns. Depending on the data size, the tracker may either set up the nodes in a tree pattern or a ring pattern.
+
+.. figure:: images/tree.png
+   :width: 80%
+   :height: 400
+   :scale: 85
+   :align: center
+   :alt: Tree Topology
+
+.. figure:: images/ring.png
+   :width: 100% 
+   :alt: Ring Topology 
+   :align: center
+
+Once the topology has been set up, each node in the cluster is assigned a rank. The rank is essentially the node identifier in the cluster. The node assigned rank 0 is considered the "master" node, and all other nodes are considered worker nodes.
+
+The tracker is logically distinct from all nodes in the cluster, and does not actually perform any computation. Once it has finished setting up the topology, the tracker has finished its job. However, a user may choose to colocate the tracker with a master/worker node for simplicity. 
+
+Before training begins, Distributed Secure XGBoost leverages *inter-enclave attestation* to authenticate all enclaves in the cluster -- each enclave attests all neighboring enclaves to verify the enclave's identity and to verify that the enclave will be running the proper code.
```

Comment:
Should we add something here about what the client attests?

Reply:
There's no client in this demo. It's just a demo of distributed training -- there's no concept of client/server as we're not doing outsourced computation.

Reply:
See comment above.